pancancer_evaluation/utilities/analysis_utilities.py

```diff
@@ -72,6 +73,42 @@ def generate_nonzero_coefficients(results_dir):
             yield identifier, coefs
 
 
+def generate_nonzero_coefficients_for_gene(results_dir, gene_name):
+    """Generate coefficients from mutation prediction model fits for a gene.
+
+    Loading all coefficients into memory at once is prohibitive, so we generate
+    them individually and analyze/summarize in analysis scripts.
+
+    Arguments
+    ---------
+    results_dir (str): directory to look in for results, subdirectories should
+                       be experiments for individual genes
+    gene (str): gene to look for
+
+    Yields
+    ------
+    identifier (str): identifier for given coefficients
+    coefs (dict): list of nonzero coefficients for each fold of CV, for the
+                  given identifier
+    """
+    coefs = {}
+    all_features = None
+    gene_dir = os.path.join(results_dir, gene_name)
+    if not os.path.isdir(gene_dir): raise StopIteration
+    for coefs_file in os.listdir(gene_dir):
+        if coefs_file[0] == '.': continue
+        if 'signal' not in coefs_file: continue
+        if 'coefficients' not in coefs_file: continue
+        cancer_type = coefs_file.split('_')[1]
```

Comment:
Magic numbers are bad, but this one is probably harder to fix than it's worth. Maybe making a note in the scripts that generate coefficient files that the cancer type must be the second field would help keep this from breaking.

Reply:
Will do! I agree the magic number isn't ideal, but as long as I use the same convention everywhere (identifier = {gene}_{cancer_type}), I don't think it will cause issues.

I'll definitely make a note in the scripts that generate these files, though.